src/components/ExpandedNode/ExpandedNode.tsx

```diff
@@ -0,0 +1,195 @@
+import React, { FC, ReactNode, ReactElement } from 'react';
+import { GuidanceState, DocumentationResource, State } from 'pathways-model';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+import classes from './ExpandedNode.module.scss';
+import indexClasses from 'styles/index.module.scss';
+import { isGuidanceState, isBranchState } from 'components/Node/Node';
+
+interface ExpandedNodeProps {
+  pathwayState: GuidanceState | State;
+  isActionable: boolean;
+  documentation: DocumentationResource | undefined;
+}
+
+const ExpandedNode: FC<ExpandedNodeProps> = ({ pathwayState, isActionable, documentation }) => {
+  const guidance =
+    isGuidanceState(pathwayState) && renderGuidance(pathwayState as GuidanceState, documentation);
+  const branch = isBranchState(pathwayState) && renderBranch(documentation);
+
+  return (
+    <div className={indexClasses.ExpandedNode}>
+      <table className={classes.infoTable}>
+        <tbody>{guidance || branch}</tbody>
+      </table>
+      {isActionable && (
+        <form className={classes.commentsForm}>
+          <label>Comments:</label>
+          <button className={indexClasses.button} onClick={(e): void => e.preventDefault()}>
+            Use Default Text
+          </button>
+          <textarea className={classes.comments}></textarea>
+          <button
+            className={`${classes.accept} ${indexClasses.button}`}
+            onClick={(e): void => e.preventDefault()}
+          >
+            Accept
+          </button>
+          <button
+            className={`${classes.decline} ${indexClasses.button}`}
+            onClick={(e): void => e.preventDefault()}
+          >
+            Decline
+          </button>
+        </form>
+      )}
+    </div>
+  );
+};
+
+type ExpandedNodeFieldProps = {
+  title: string;
+  description: ReactNode;
+};
+
+const ExpandedNodeField: FC<ExpandedNodeFieldProps> = ({ title, description }) => {
+  return (
+    <tr>
+      <td className={classes.descTitle}>{title}</td>
+      <td className={classes.desc}>{description}</td>
+    </tr>
+  );
+};
+
+function renderBranch(documentation: DocumentationResource | undefined): ReactElement[] {
+  const returnElements: ReactElement[] = [];
+
+  if (documentation?.resource) {
+    switch (documentation.resourceType) {
+      case 'Observation': {
+        const observation = documentation.resource as fhir.Observation;
+
+        const valueCoding = observation.valueCodeableConcept?.coding;
+        valueCoding &&
+          returnElements.push(
+            <ExpandedNodeField
+              key="ValueSystem"
```

Comment:
We probably want to use unique keys for these components.  If we have multiple branches expanded, these components would all have the same key value.  Maybe add the `pathwayState` label to the keys? I would change all the non-unique keys in `renderBranch` and `renderGuidance`.

Reply:
I believe react keys only need to be unique within the component rendering them. In this case this is the ExpandedNode component which calls `renderBranch` and `renderGuidance`. Since multiple `ExpandedNodeField`s are rendered these all need a key. However I do not believe they need to be unique between nodes. To ensure they are unique within the `ExpandedNode` I prepended the keys for fields related to the value with "Value". This way instead of having two keys named "System" we instead have "System" and "ValueSystem". I believe this is enough, but correct me if I am misunderstanding

Reply:
Ahh cool I think you're right and this should be fine.  I always thought they needed globally unique.